### PR TITLE
[apex] Whitelisting String.isEmpty and casting  

### PR DESCRIPTION
**Pull request by sgorbaty**
Hi @jsotuyod,

This is to reduce FPs when URL param variable passed to String.isEmpty.

Cheers,
Sergey

### CODE REVIEW

**Comment by jsotuyod** [comment]
@sgorbaty this is missing a test case showing the FP. Could you please provide one? That was we can be sure the issue existed, doesn't exist anymore, and won't come back in the future.

Thanks

**Comment by sgorbaty** [comment]
@jsotuyod you got it! Actually, I found a way to make it even better!

File: pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/security/xml/ApexXSSFromURLParam.xml [hunk 1]
```diff
@@ -275,6 +295,32 @@ public class Foo {
...
+		]]></code>
+	</test-code>
+
+	<test-code>
+		<description>URL parameter type empty check</description>
+		<expected-problems>0</expected-problems>
+		<code><![CDATA[
+public class Foo {
+	public void test1() {
+		boolean isEmpty = String.isEmpty(ApexPages.currentPage().getParameters().get('foo'););
```
Comment (jsotuyod) [inline pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/security/xml/ApexXSSFromURLParam.xml]: this line has a misplaced `;` doesn't it?

**Comment by jsotuyod** [comment]
This will be included in PMD 5.5.3 and later

**Pull request merged**